app/storage/questionnaire_state_database_storage.py

```diff
@@ -0,0 +1,57 @@
+import logging
+
+from app.data_model.database import QuestionnaireState, commit_or_rollback
+from app.data_model.database import db_session
+
+logger = logging.getLogger(__name__)
+
+
+class QuestionnaireDatabaseStorage:
+    """
+    Server side storage using an RDS database (where one column is the entire JSON representation of the questionnaire state)
+    """
+
+    def __init__(self, user_id):
+        if user_id is None:
+            raise ValueError('User id must be set')
+        self.user_id = user_id
+
+    def add_or_update_questionnaire_state(self, data):
+        """
+        If users data does not exist add it otherwise update the existing data.
+        :param data: users data to add or update
+        :return:
+        """
+        logger.debug("About to store data %s for user %s", data, self.user_id)
+        if self.has_questionnaire_state():
+            logger.debug("Updating data for user %s", self.user_id)
+            questionnaire_state = self._get_questionnaire_state()
+            questionnaire_state.set_data(data)
+        else:
+            logger.debug("Creating questionnaire state for user %s with data %s", self.user_id, data)
+            questionnaire_state = QuestionnaireState(self.user_id, data)
+
+        with commit_or_rollback(db_session):
+            db_session.add(questionnaire_state)
+
+    def get_user_data(self):
+        logger.debug("Loading questionnaire state for user %s", self.user_id)
+        questionnaire_state = QuestionnaireState.query.filter(QuestionnaireState.user_id == self.user_id).first()
+        return questionnaire_state.get_data()
+
+    def _get_questionnaire_state(self):
```

Comment:
Surely we've qualified the type of data by the class we're using, do we really need the questionnaire_state() suffixed to each method?